scripts/unist.d.ts

```diff
@@ -17,46 +17,46 @@ export type FlowNode =
   | PreNode
   | CodeNode;
 
-export interface BlockquoteNode extends Parent {
+export type BlockquoteNode = Parent & {
   type: "blockquote";
   children: FlowNode[];
-}
+};
 
-export interface HeadingNode extends Parent {
+export type HeadingNode = Parent & {
   type: "heading";
   depth: number;
   children: PhrasingNode[];
-}
+};
 
-export interface ParagraphNode extends Parent {
+export type ParagraphNode = Parent & {
   type: "paragraph";
   children: PhrasingNode[];
-}
+};
 
-export interface PreNode extends Parent {
+export type PreNode = Parent & {
   type: "pre";
   children: PhrasingNode[];
-}
+};
 
-export interface CodeNode extends Parent {
+export type CodeNode = Parent & {
   type: "code";
   children: PhrasingNode[];
   value?: string;
   meta?: string | string[];
-}
+};
 
-export interface EmphasisNode extends Parent {
+export type EmphasisNode = Parent & {
   type: "emphasis";
   children: PhrasingNode[];
-}
+};
 
-export interface LinkNode extends Parent {
+export type LinkNode = Parent & {
   type: "link";
   children: FlowNode[];
   url?: string;
-}
+};
 
-export interface TextNode extends Literal {
+export type TextNode = Literal & {
   type: "text";
   value: string;
-}
+};
```

Comment:
FWIW intersection types apparently perform far worse than `interface / extends`:

https://twitter.com/mattpocockuk/status/1698642590193619025

Might not matter at Remix's current size, but probably something to be aware of as the codebase grows

Reply:
Agreed - this is why I want us to define/document our team stance before we go making these types of changes.  I'm going to close this for now and we'll get a TS-cleanup on the books.